### DISCUSSION TITLE
Road Map - Long Term Bar -  When does it become zero

### DISCUSSION THREAD

**Discussion by rsrinivasanhome** [simonec73/threatsmanager#11] [Q&A] [answered]
Reference :  https://github.com/simonec73/threatsmanager/discussions/10#discussioncomment-1498089

For test purposes for all threat events I have added a mitigation of strength maximum.
I have planned all mitigations to  be completed  by mid term.
My long term bar still does not come to zero why ?  Will the long term bar ever come to zero ?

![image](https://user-images.githubusercontent.com/31912418/138984262-2569569b-7247-4e6b-9cfb-2290e5b8282f.png)

![image](https://user-images.githubusercontent.com/31912418/138984318-fb95bc4b-be4f-4b5f-98c1-a2b628645ec7.png)

In the below feature what is functionality of the create item button 
https://threatsmanager.com/training/advanced/concepts/merge-threat-models-and-templates/

My assumption is that mitigations under the "No Action Required "  bucket will not influence the bar chart .  Am I correct ?

On another note at times templates create false positives .  What is the recommended approach to handle them (other than updating the template) ? Should we delete those threat events  ? @simonec73 can you provide your guidance here ?

I want to merge two templates which have two objects which have the same stencil type Example I have two definitions of "Native Application" process with different  properties .  
Will the merge tool be able to merge the additional properties together and not create two duplicate objects.

I wanted to create a backlog for my pentest team based on my threat model .  Will the review notes feature help for  this ?

I was going deep into the xml created by microsoft threat modeler and was wonder how you are identifying the stencil is a "Data Store" , Process or "External Interactor"  .  When I import the template - https://github.com/AzureArchitecture/threat-model-templates  it look like azure adb2c is coming as external interactor .  I feel  Azure ADB2C should be a processes.

**Comment by simonec73** [comment, marked as answer]
Sure! And thank you again for keeping the conversation going.

The color of the bars is the result of two factors:
- The length of the bar, which is determined by the severity of the Threat Events. The Current bar shows the current Severities. If you have all Severities to Info then the overall value would be extremely low. The length of the other bars is determined by the current severities and by the mitigations you plan in each phase. This means that typically you will have decreasing lengths, because the short term bar is determined by the current severities and by the mitigations you have planned for the short term phase, then the mid-term bar is determined by the current severities and by the mitigations planned for both short term and mid-term phases, and finally, the long term bar is determined by the current severities and by the mitigations planned for all the three phases.
- The color of the bars is determined by the parameters used for the Bug Bar Residual Risk Estimator. I see that all four bars are green: this means that you have chosen pretty high values for the parameters. Try lowering them.
![image](https://user-images.githubusercontent.com/23357188/139588353-30a8c64a-41a6-47a1-bb51-29b4550835e4.png)
Notice that the values you specify may be normalized to a reference model. You can see that in the Residual Risk Estimator Configuration dialog (see above): it's clearly stated in the text at the bottom of the dialog. The 30 objects shown in the picture include all entities and flows. If that happens, and you have a very small model, then this may cause relatively small numbers to increase their effectiveness, and as a result, you see more green than you should. The goal of this feature is to allow using the same parameters even if your threat model increases in size, which typically happens when you are applying threat modeling to an Agile project, spanning multiple cycles. This is configured in the Options.
![image](https://user-images.githubusercontent.com/23357188/139588596-aff1fcfc-face-4fa2-b26b-c3fae4fe43c6.png)

In your specific case, I guess that the culprit is simply due to the fact that you are using the Threat Event Severity wrongly: this should represent the current severity, not what you have at the end of the process. In other words, the Severity you specify would be the result of the original Severity, decremented by the effect of all the Existing and Implemented Mitigations. In fact, if you do that, your Mitigations in any other state would not be effective in decreasing the risk, because it's already very low!

Another mistake I see is that all mitigations are Existing and Implemented:
- Existing mitigations mean that you found them. They are not the result of your work as a threat modeling expert.
- Implemented mitigations are those which you identified during a first iteration of the threat model, and that the team has implemented. Now, you mark them as Implemented during a follow-up revision of the threat model.
Please, get a look at https://threatsmanager.com/training/intro/basic/mitigations/ for a description of the various states.

For the Create Item, ok I see you are referring to the Merge. The idea is that the Merge tool shows you all the differences between the current Threat Model and a Model or Template of your choice. The button you are referring you will simply show what would happen if you decide to Merge a specific object. Create Item is shown only when the object would be added to the current Threat Model, because it's missing. If you had an Existing object with differences, instead of reading Create Item, you would read what would be changed during the Merge.

No Action Required in the Roadmap: correct. Those are mitigations which have been identified as not requiring any action, because they are applied to threat events and for all of them the status is Implemented or Existing (therefore, you are set), or because you specifically decide that the mitigation is not recommended. In the last case, please remember to add a note using the Annotations (Quality extension), to specify why this is not required.

False positives. You have various options:
- If they are simply wrong, then you can remove the false positive (Threat Event or Mitigation).
- If someone may think they would apply, but really not in this case, then you can maintain them. Threat Events in this condition would need to be marked as Info severity, and Mitigations should be marked as having a Negligible strength.
- If the Threat Event would apply, but is fully mitigated, then change its severity to Info. Remember to check that the applied mitigations are listed and marked as Existing or Implemented.

About merging two objects: if they have the same ID, then they would be merged, otherwise two objects would be created. You can see the ID in Expert or Pioneer mode. Please consider that the merge has limits, and existing properties would not be replaced. The upcoming version (1.5.0) improves over that.

Review notes and pentest. Well, not really at this point, because they are not going to be exported with the mitigation. That's an interesting use case I didn't consider. Can we talk a little? I may be able to work with you on this and provide the required feature with the new version.

About the Import functionality: If something is marked as an External Interactor in the.TB7, then it is imported as an External Interactor. I may agree that in some cases you may want ADB2C as a Process (that's a pretty rare situation, though, based on my experience), but my tool does not make decisions. Change the source template to get a Process instead of an interactor.

**Reply by rsrinivasanhome** [reply]
First of all, would like to thank you for your time.  The responses have been very insightful.  There is one area where I still require help.
I still do not understand when the road map bars will come to zero .  I have been setting the migration status to Existing or Implemented to test when the height of the road map bar becomes zero . 
I have made the severity of all threat events as Info and still see roadmap bars do not become zero.  
If my threats have been mitigated (migration status existing or implemented) and remaining threat event - severity as info  then what is the residual risk ?  Why do the bars not come down to zero ?

**Reply by simonec73** [reply]
Thank you for your question.

Simply put: if you have at least a threat event, it cannot happen, and this is by design. In fact, even if you manage to fully mitigate all threat events, they still would have Info severity, which means 1pt.
This is in line with the consideration that the only way to make the risk zero is to remove the function causing that risk. In fact, every mitigation will only reduce the risk for some extent, hopefully.
There is a lot more to tell about risks and mitigations: for example, get a look at https://simoneonsecurity.com/2020/02/27/mitigations-oh-my/.
I hope it helps.
